internal/wal_push_handler.go

```diff
@@ -50,6 +80,12 @@ func HandleWALPush(uploader *WalUploader, walFilePath string) {
 	bgUploader.Start()
 	err = uploadWALFile(uploader, walFilePath, bgUploader.preventWalOverwrite)
 	tracelog.ErrorLogger.FatalOnError(err)
+	if err == nil && (viper.IsSet(UploadWalMetadata) || viper.GetString(UploadWalMetadata) == WalNoMetadataLevel) {
```

Comment:
I guess here you check instead for `viper.GetString(UploadWalMetadata) != WalNoMetadataLevel `?

Actually, let's just set the default value for UploadWalMetadata to "NOMETADATA" to avoid viper.IsSet check.
This can be done in config.go:108